server.js

```diff
@@ -0,0 +1,157 @@
+    // Remove default x-powered-by response header
+    app.disable("x-powered-by");
+
+    // Prevent opening page in frame or iframe to protect from clickjacking
+    app.use(helmet.frameguard()); //xframe deprecated
+
+    // Prevents browser from caching and storing page
+    app.use(helmet.noCache());
+
+    // Allow loading resources only from white-listed domains
+    app.use(helmet.contentSecurityPolicy()); //csp deprecated
+
+    // Allow communication only on HTTPS
+    app.use(helmet.hsts());
+
+    // TODO: Add another vuln: https://github.com/helmetjs/helmet/issues/26
+    // Enable XSS filter in IE (On by default)
+    // app.use(helmet.iexss());
+    // Now it should be used in hit way, but the README alerts that could be
+    // dangerous, like specified in the issue.
+    // app.use(helmet.xssFilter({ setOnOldIE: true }));
+
+    // Forces browser to only use the Content-Type set in the response header instead of sniffing or guessing it
+    app.use(nosniff());
+    */
+
+    // Adding/ remove HTTP Headers for security
+    app.use(favicon(__dirname + "/app/assets/favicon.ico"));
+
+    // Express middleware to populate "req.body" so we can access POST variables
+    app.use(bodyParser.json());
+    app.use(bodyParser.urlencoded({
+        // Mandatory in Express v4
+        extended: false
+    }));
+
+    // Enable session management using express middleware
+    app.use(session({
+        // genid: (req) => {
+        //    return genuuid() // use UUIDs for session IDs
+        //},
+        secret: cookieSecret,
+        // Both mandatory in Express v4
+        saveUninitialized: true,
+        resave: true
+        /*
+        // Fix for A5 - Security MisConfig
+        // Use generic cookie name
+        key: "sessionId",
+        */
+
+        /*
+        // Fix for A3 - XSS
+        // TODO: Add "maxAge"
+        cookie: {
+            httpOnly: true
+            // Remember to start an HTTPS server to get this working
+            // secure: true
+        }
+        */
+
+    }));
+
+    /*
+    // Fix for A8 - CSRF
+    // Enable Express csrf protection
+    app.use(csrf());
+    // Make csrf token available in templates
+    app.use((req, res, next) => {
+        res.locals.csrftoken = req.csrfToken();
+        next();
+    });
+    */
+
+    // Register templating engine
+    app.engine(".html", consolidate.swig);
+    app.set("view engine", "html");
+    app.set("views", `${__dirname}/app/views`);
+    // Fix for A5 - Security MisConfig
+    // TODO: make sure assets are declared before app.use(session())
+    app.use(express.static(`${__dirname}/app/assets`));
+
+
+    // Initializing marked library
+    // Fix for A9 - Insecure Dependencies
+    marked.setOptions({
+        sanitize: true
+    });
+    app.locals.marked = marked;
+
+    // Application routes
+    routes(app, db);
+
+    // Template system setup
+    swig.setDefaults({
+        // Autoescape disabled
+        autoescape: false
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- 2493504a2479e66ca14e1a9a6cb93751 -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Improper Encoding or Escaping of Output
_Medium CWE-116_
Setting the `autoescape` attribute to `false` disables the escaping mechanism of the Swig template engine. This may lead to XSS attack if untrusted data is not properly validated and escaped.



### How to fix
Potential mitigations include the following:
  1. Use a vetted library or framework that does not allow this weakness to occur or provides constructs that make this weakness easier to avoid.

For example, consider using the ESAPI Encoding control [REF-45] or a similar tool, library, or framework. These will help the programmer encode outputs in a manner less prone to error.

Alternately, use built-in functions, but consider using wrappers in case those functions are discovered to have a vulnerability.

  2. If available, use structured mechanisms that automatically enforce the separation between data and code. These mechanisms may be able to provide the relevant quoting, encoding, and validation automatically, instead of relying on the developer to provide this capability at every point where output is generated.

For example, stored procedures can enforce database query structure and reduce the likelihood of SQL injection.

  3. Understand the context in which your data will be used and the encoding that will be expected. This is especially important when transmitting data between different components, or when generating outputs that can contain multiple encodings at the same time, such as web pages or multi-part mail messages. Study all expected communication protocols and data representations to determine the required encoding strategies.
  4. In some cases, input validation may be an important strategy when output encoding is not a complete solution. For example, you may be providing the same output that will be processed by multiple consumers that use different encodings or representations. In other cases, you may be required to allow user-supplied input to contain control information, such as limited HTML tags that support formatting in a wiki or bulletin board. When this type of requirement must be met, use an extremely strict allowlist to limit which control sequences can be used. Verify that the resulting syntactic structure is what you expect. Use your normal encoding methods for the remainder of the input.
  5. Use input validation as a defense-in-depth measure to reduce the likelihood of output encoding errors (see CWE-20).
  6. Fully specify which encodings are required by components that will be communicating with each other.
  7. When exchanging data between components, ensure that both components are using the same character encoding. Ensure that the proper encoding is applied at each interface. Explicitly set the encoding you are using whenever the protocol allows you to do so.